test/rules/assertions/checkParamNames.js

```diff
@@ -1835,5 +1856,20 @@ export default {
         parser: typescriptEslintParser
       },
     },
+    {
+      code: `
+        /**
+         * @param foo
+         * @param foo.bar
+         */
+        function quux (bar, foo) {
+        }
+      `,
+      options: [
+        {
+          disableMissingParamChecks: true,
```

Comment:
Is there a test case we can add where `disableMissingParamChecks: true` but which still fails?

Reply:
Yea, we only want to skip known params so this should fail. The implementation is way too naive.

```
        /**
         * @param foo
         * @param foo.bar
         */
        function quux (bar, baz) {
        }
```

Reply:
I've added two new test cases. If I understand correctly, that code is not hit when the expected or actual names include property lookup for objects or indices so it seems to be a nice and simple solution.

Reply:
How about another test where the params are `foo, bar` and the only value present is `foo`?

Reply:
Of course. I ended up adding 2:
* Your suggestion - When the option is set it is invalid if there is an extra `@param`
* When the option is set it is valid to have no tags

And moved the last one:
* When the option is set it is valid to skip a tag